python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -90,6 +97,15 @@ def execute_job_backfill_iteration(
             yield None
             time.sleep(CHECKPOINT_INTERVAL)
         else:
+            backfill_runs = instance.get_runs(
+                RunsFilter(tags=DagsterRun.tags_for_backfill_id(backfill.backfill_id))
+            )
+            for run in backfill_runs:
+                if run.status not in FINISHED_STATUSES:
```

Comment:
still figuring out how to test this behavior change. The unit tests use the in process executor so the runs are always finished by the time we get to this point in the daemon